### DISCUSSION TITLE
What mechanism is used by ccache to purge files?

### DISCUSSION THREAD

**Discussion by rraallvv** [ccache/ccache#207] [Questions & answers] [answered]
I tried ccache with a project that uses SCons build system before realizing that SCons has it's own cache system. So I'm curios about what mechanism is used by ccache to purge files.

Currently I'm using [this handler](https://bitbucket.org/scons/scons/wiki/LimitCacheSizeWithProgress) suggested on the SCons wiki page. It removes those files with the smallest size first, with the caveat that the size decays exponentially with time.

Does ccache have a similar approach?

**Comment by afbjorklund** [comment, marked as answer]
You can find the sorting code in the clean routines, it uses age (least recent modification time) to clean.

https://github.com/ccache/ccache/blob/master/cleanup.c

**Comment by rraallvv** [comment]
@afbjorklund hmm... I ended up doing the same thing, I'm just wondering whether using the file size with exponential decay for sorting the files is really being used somewhere in production environments, or whether that is just some theoretical thing. Thanks for the info.
